### ISSUE TITLE
`mallocFresh` unsound due to blob non-uniqueness

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#1070]
While working on #1069, there was some behavior which seemed suspect. Namely that `p->a = q->b` style assignments where `p` and `q` point to the same fresh blob should not cause it to escape, but remain fresh, such that no race is reported. This is a pattern from the silver searcher.

But this style of exclusion is not sound since there's no guarantee that `p` and `q` point to the same physical memory, we just have the same `alloc` `varinfo` for it. #1069 is not at fault here, the issue exists already in the `master` version of `mallocFresh`.

The analysis is sound with `region`, from which `mallocFresh` was extracted, because `region` does not even distinguish allocation sites, but simply uses bullet for all of them. `mallocFresh` was made slightly more precise to handle some interactive story by maintaining a set of fresh `alloc` variables, but already this change is enough to turn it unsound.

### Example
Should report a race on the `next` field, but does not.

```c
// PARAM: --set ana.activated[+] mallocFresh --set ana.activated[-] mhp --set ana.thread.domain plain
#include <pthread.h>
#include <stdlib.h>

struct s {
  struct s *next;
};

struct s *g = NULL;
pthread_mutex_t A = PTHREAD_MUTEX_INITIALIZER;

void *t_fun(void *arg) {
  pthread_mutex_lock(&A);
  struct s *r = g->next; // RACE!
  pthread_mutex_unlock(&A);
  return NULL;
}

struct s* alloc_s() {
  return malloc(sizeof(struct s));
}

int main() {
  struct s *p, *q;
  int r;

  pthread_t id;
  pthread_create(&id, NULL, t_fun, NULL);

  p = alloc_s();
  // p->next = NULL; // NORACE (fresh)
  pthread_mutex_lock(&A);
  g = p; // NORACE
  pthread_mutex_unlock(&A);

  q = alloc_s(); // UNSOUND: same alloc varinfo without wrapper becomes fresh again!
  // q->next = NULL; // NORACE (fresh)

  // p and q point to same blob, so accessing via both should race
  if (r)
    p->next = q; // RACE!
  else
    g->next = q; // RACE!
  return 0;
}
```
